src/net_processing.cpp

```diff
@@ -4607,11 +4607,54 @@ void PeerManagerImpl::ConsiderEviction(CNode& pto, int64_t time_in_seconds)
 
 void PeerManagerImpl::EvictExtraOutboundPeers(int64_t time_in_seconds)
 {
-    // Check whether we have too many outbound peers
-    int extra_peers = m_connman.GetExtraOutboundCount();
-    if (extra_peers > 0) {
-        // If we have more outbound peers than we target, disconnect one.
-        // Pick the outbound peer that least recently announced
+    // If we have any extra block-relay-only peers, disconnect the youngest unless
```

Comment:
19858: missing potentially useful changes in `PeerManager::ProcessMessage` (less merge conflicts in the future)
```diff
diff --git a/src/net_processing.cpp b/src/net_processing.cpp
index 10d192a0c3..66334d1532 100644
--- a/src/net_processing.cpp
+++ b/src/net_processing.cpp
@@ -3097,7 +3097,7 @@ void PeerManagerImpl::ProcessMessage(
             LogPrintf("New outbound peer connected: version: %d, blocks=%d, peer=%d%s (%s)\n",
                       pfrom.nVersion.load(), pfrom.nStartingHeight,
                       pfrom.GetId(), (fLogIPs ? strprintf(", peeraddr=%s", pfrom.addr.ToString()) : ""),
-                      pfrom.RelayAddrsWithConn()?  "full-relay" : "block-relay");
+                      pfrom.IsBlockOnlyConn() ? "block-relay" : "full-relay");
         }
 
         if (!pfrom.m_masternode_probe_connection) {
```